openhands/server/settings.py

```diff
@@ -38,22 +45,53 @@ def llm_api_key_serializer(self, llm_api_key: SecretStr, info: SerializationInfo
 
         return pydantic_encoder(llm_api_key)
 
-    @field_serializer('github_token')
-    def github_token_serializer(
-        self, github_token: SecretStr | None, info: SerializationInfo
-    ):
-        """Custom serializer for the GitHub token.
-
-        To serialize the token instead of ********, set expose_secrets to True in the serialization context.
-        """
-        if github_token is None:
-            return None
-
-        context = info.context
-        if context and context.get('expose_secrets', False):
-            return github_token.get_secret_value()
-
-        return pydantic_encoder(github_token)
+    @model_validator(mode='before')
+    @classmethod
+    def convert_provider_tokens(cls, data: dict) -> dict:
+        """Convert provider tokens from JSON format to SecretStore format."""
+        if isinstance(data, dict):
+            if 'secrets_store' in data and isinstance(data['secrets_store'], dict):
+                # Handle direct provider_tokens in secrets_store
+                if 'provider_tokens' in data['secrets_store']:
+                    tokens = data['secrets_store']['provider_tokens']
+                    if isinstance(tokens, dict):
+                        # Create a new SecretStore with the tokens
+                        converted_tokens = {}
+                        for token_type_str, token_value in tokens.items():
+                            if token_value:
+                                try:
+                                    # Convert string to ProviderType enum
+                                    token_type = ProviderType(token_type_str)
+                                    # Handle both string and dict token values
+                                    if isinstance(token_value, dict):
+                                        token_str = token_value.get('token')
+                                        if token_str:
+                                            converted_tokens[token_type] = (
+                                                ProviderToken(
```

Comment:
lots of nesting here--can we return/continue/break to avoid?

Reply:
@openhands there is a lot of nesting for  `convert_provider_tokens`

can we make it more readable using return/continue/break